esp-wifi/src/preempt/mod.rs

```diff
@@ -22,7 +22,7 @@ pub fn allocate_main_task() -> *mut Context {
         }
 
         let ptr = malloc(size_of::<Context>() as u32) as *mut Context;
-        core::ptr::write_bytes(ptr, 0, 1);
+        core::ptr::write(ptr, Context::new());
```

Comment:
This really should be `Box::new` but I'm not ready to go down that rabbit hole yet.

Reply:
Ideally we would do construct-in-place